libs/cart/src/drivers/interfaces/cart-coupon-service.interface.ts

```diff
@@ -0,0 +1,29 @@
+import { InjectionToken } from '@angular/core';
+import { Observable } from 'rxjs';
+
+import { DaffCartCoupon } from '../../models/cart-coupon';
+import { DaffCart } from '../../models/cart';
+
+/**
+ * The interface responsible for applying a coupon to a cart.
+ */
+export interface DaffCartCouponServiceInterface<T extends DaffCart = DaffCart, V extends DaffCartCoupon = DaffCartCoupon> {
```

Comment:
I think we need to be able to `list` all the coupons applied to the cart as well.

Reply:
wouldn't you just do a getCart call? To list the coupons I think you'd usually need to make a getCart query that only returns the coupons

Reply:
I can't think of a scenario where you'd want the coupons applied to the cart without the cart itself.

Reply:
Maybe if you had a checkout page just for the purpose of listing coupons and discounts? The other cart drivers have methods for getting the specific models so I think a `list` method would be consistent.

Reply:
I don't think there's even a way to query for coupons that aren't associated with a cart. I think that's because you don't typically want everyone to know about all the coupon codes you're offering. I think that online coupons are usually used by people who are promoting your product to get people to visit the site, so that the seller can credit the promoter for bringing the customer to the site. Discounts might be a different story. Like maybe you could filter products by discounted ones.

Reply:
@lderrickable I mean listing coupons on the cart. We provide `get` methods for other cart submodels and its nice to have a smaller query when you don't need the entire cart.

Reply:
My original point was what @griest024 is describing.

Reply:
LGB actively advertises promo codes on their website, everywhere. So it might actually be helpful to have a 'View Promo Codes' modal on the cart page by the promo code solo field that displays all the available promo codes. This is out of scope of this PR though and dependents on the client.

But I feel like you guys are discussing two different functionalities? Listing all available coupons vs listing applied coupons?

The latter, listing applied coupons is required. It's the same idea as listing applied filters. A user should be able to see and remove applied coupons.

Reply:
ok I'll add it

Reply:
Stores often actively advertise promo codes on their website (i.e. Sephora has a separate page listing their promos). So it might actually be helpful to have a 'View Promo Codes' modal on the cart page by the promo code solo field that displays all the available promo codes. This is out of scope of this PR though.

But I feel like you guys are discussing two different functionalities? Listing all available coupons vs listing applied coupons?

The latter (listing applied coupons) is required anywhere adding a promo code is an option (cart, checkout, cart sidebar). A user should be able to see and remove applied coupons.

Reply:
As far as I can tell, there is no way to list coupon codes by themselves apart from those actively applied to a cart. I looked at Sephora's website, and I suspect they have that page semi-hardcoded. The promo codes are actually shown as static images.

Reply:
added

Reply:
Yeah I think listing all coupons codes and promos that exist for a site is more like an admin panel thing than a storefront action. Definitely out of this PR's scope anyway.